dapps/v2Explorer/src/components/ExplorerModal.tsx

```diff
@@ -0,0 +1,178 @@
+import React, {useCallback, useEffect, useState} from 'react';
+import {StyleSheet, View, useColorScheme, Dimensions} from 'react-native';
+import Modal from 'react-native-modal';
+import {InitialExplorerContent} from './InitialExplorerContent';
+import {ViewAllExplorerContent} from './ViewAllExplorerContent';
+
+//@ts-expect-error - `@env` is a virtualised module via Babel config.
+import {fetchInitialWallets, fetchViewAllWallets} from '../utils/ExplorerUtils';
+import {ExplorerModalHeader} from './ExplorerModalHeader';
+
+const MODAL_HEIGHT = Dimensions.get('window').height * 0.7;
+const DEVICE_WIDTH = Dimensions.get('window').width;
+
+interface ExplorerModalProps {
+  modalVisible: boolean;
+  close: () => void;
+}
+
+// Populate with the data...
+export function ExplorerModal({modalVisible, close}: ExplorerModalProps) {
+  const [isLoading, setIsLoading] = useState(true);
+  const [isViewAllLoading, setViewAllLoading] = useState(true);
+
+  const [viewAllContentVisible, setViewAllContentVisible] = useState(false);
+  const [explorerData, setExplorerData] = useState([]);
+  const [viewAllExplorerData, setViewAllExplorerData] = useState([]);
+
+  const isDarkMode = useColorScheme() === 'dark';
+
+  const loading = (loadingState: boolean) => {
+    setIsLoading(loadingState);
+  };
+
+  const loadingViewAll = (loadingState: boolean) => {
+    setViewAllLoading(loadingState);
+  };
+
+  const fetchExplorerData = (data: any) => {
+    setExplorerData(data);
+  };
+
+  const fetchViewAllExplorerData = (data: any) => {
+    setViewAllExplorerData(data);
+  };
+
+  const fetchWallets = useCallback(() => {
+    fetchInitialWallets(loading, fetchExplorerData);
+    fetchViewAllWallets(loadingViewAll, fetchViewAllExplorerData);
+  }, []);
+
+  useEffect(() => {
+    if (!explorerData.length) {
+      fetchWallets();
+    }
+  }, [explorerData, fetchWallets, isLoading, isViewAllLoading]);
+
+  return (
+    <Modal
+      isVisible={modalVisible}
+      onModalHide={() => setViewAllContentVisible(false)}
+      useNativeDriver>
+      <View style={styles.wcContainer}>
+        <ExplorerModalHeader close={close} />
+        <View
+          style={
+            isDarkMode
+              ? styles.connectWalletContainer
+              : styles.connectWalletContainerLight
+          }>
+          {!viewAllContentVisible ? (
+            <InitialExplorerContent
+              isLoading={isLoading}
+              explorerData={explorerData}
+              setViewAllContentVisible={setViewAllContentVisible}
+            />
+          ) : (
+            <ViewAllExplorerContent
+              explorerData={viewAllExplorerData}
+              setViewAllContentVisible={setViewAllContentVisible}
+            />
+          )}
+        </View>
+      </View>
+    </Modal>
+  );
+}
+
+const styles = StyleSheet.create({
+  wcContainer: {
+    // flex: 1,
+    position: 'absolute',
+    bottom: -20,
+    left: -20,
+    // maxHeight: MODAL_HEIGHT,
```

Comment:
Leftover commented styles. Should be removed unless they serve an optional purpose.